File: go/1.go
// Author: Andrew Hamara

// Solution to leetcode problem 1. Two Sum

func twoSum(nums []int, target int) []int {

    visited := make(map[int]int)

    for i:=0; i < len(nums); i++ {
        
        var cur int = target - nums[i]

        val, ok := visited[cur]

        if ok {
            return []int{i, val}
        }

        visited[nums[i]] = i
    }

    return nil
}

File: py/1.py
# Author: Andrew Hamara

# Solution to LeetCode problem 1. Two Sum

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        visited = {}
        for i, num in enumerate(nums):
            temp = target - num
            if temp in visited:
                return [visited[temp], i]
            visited[num] = i
